src/models/streaming/BufferedDuplex.ts

```diff
@@ -0,0 +1,43 @@
+import { Duplex, PassThrough } from 'stream';
+
+export class BufferedDuplex extends Duplex {
+    input: PassThrough;
+    output: PassThrough;
+    areOutputHandlersSetup: boolean;
+    constructor (input: PassThrough, output: PassThrough, options?: any) {
+        super(options);
+        this.input = input;
+        this.output = output;
+        this.areOutputHandlersSetup = false;
+    }
+
+    _write(chunk: any, encoding: any, callback: any): boolean {
+        return this.input.write(chunk, encoding, callback);
+    }
+
+    _read(size: any): any {
+        if (!this.areOutputHandlersSetup){
```

Comment:
I'm not familiar with how exactly `Duplex` operates. If we setup outputhandlers only when _read is invoked are there cases when this would be too late to read incoming messages?



also minor spacing edit
```suggestion
        if (!this.areOutputHandlersSetup) {
```

Reply:
I think what you're saying is possible. But I would blame the user in that case. If they haven't set up to read from the output buffer before it gets totally full then it's on them. I'm also not sure what a good solution to this would be. Since setting it up requires reading from it in the first place